lang/src/OTS/IMP/org/partiql/ots/legacy/types/VarcharType.kt

```diff
@@ -1,13 +1,22 @@
 package OTS.IMP.org.partiql.ots.legacy.types
 
-import OTS.ITF.org.partiql.ots.TypeParameters
 import OTS.ITF.org.partiql.ots.type.ScalarType
+import OTS.ITF.org.partiql.ots.type.TypeParameters
 import org.partiql.lang.eval.ExprValue
 import org.partiql.lang.eval.ExprValueType
 
 object VarcharType : ScalarType {
-    override val id: String
-        get() = "varying_character"
+    override val id = "character_varying"
+
+    override val names = listOf("varchar", "character varying")
+
+    override fun validateParameters(typeParameters: TypeParameters) {
```

Comment:
This is interesting. Opinion, but I think it would be cleaner to have an interface for ParametricType rather than NonParametricType. My opinion, but it seems a little backwards to have the "simple" non-parameterized types to be declared specially - inheriting from an abstract class, rather than treating the parameterized "special" types as special.

Here's a simple example of what I'm suggesting

```kotlin
interface ScalarType {

    fun validateParameters(): Boolean = true

}

interface ParametricType : ScalarType {

    // Declare override without a body; forcing implementation
    override fun validateParameters(): Boolean
}

// No additional classification
class IntType : ScalarType


class VarcharType : ParametricType {
    
    override fun validateParameters(): Boolean {
        TODO()
    }

}

```